code-ready/crc
# Kill daemon process if running with a different version

## Solution/Idea

This pr try to kill an existing daemon process if it is running with different version than expected. It will allow setup to run
required version of daemon.

## Testing

- Start `crc daemon` with older version
- Try `crc setup` with this PR (bump the version to something like 2.2.0)
- `crc setup` should kill the existing `crc daemon` process and start the daemon with correct version.

Also
- Start `crc daemon` with same version
- Try `crc setup` with this PR
- `crc setup --check-only` should pass with using existing daemon process.


This will be useful for upcoming releases where user try to upgrade crc but daemon might be running with older version of crc so that case `crc setup` should kill that existing process and start daemon process with launchd/task manager.
